content/2015.09.kinto-encryption-example.rst

```diff
@@ -0,0 +1,207 @@
+  };
+
+Next, we need to create a <code>RemoteTransformer</code> that can encode and decode. And for
+that, we need to generate an AES key. We will use that key to encrypt and decrypt
+the data that is passed through the <code>RemoteTransformer</code>:
+
+.. code-block:: javascript
+  // RemoteTransformer:
+  const generateAesKey = () => {
+    // See http://www.w3.org/TR/WebCryptoAPI/#examples-symmetric-encryption
+    return window.crypto.subtle.generateKey({ name: 'AES-CBC', length: 128 },
+         false, ['encrypt', 'decrypt']);
+  };
+
+  const createTransformer = (aesKey) => {
+    const encode = (record) => {
+      const cleartext = rawStringToByteArray(JSON.stringify(record));
+      const IV = window.crypto.getRandomValues(new Uint8Array(16));
+
+      return window.crypto.subtle.encrypt({ name: 'AES-CBC', iv: IV }, aesKey,
+          cleartext).then(ciphertext => {
+        return {
+          id: record.id,
+          ciphertext: byteArrayToBase64String(new Uint8Array(ciphertext)),
+          IV: byteArrayToBase64String(IV)
+        };
+      });
+    };
+
+    const decode = (record) => {
+      const ciphertext = base64StringToByteArray(record.ciphertext);
+      const IV = base64StringToByteArray(record.IV);
+
+      return crypto.subtle.decrypt({ name: 'AES-CBC', iv: IV }, aesKey,
+          ciphertext).then(recordArrayBuffer => {
+
+        return JSON.parse(String.fromCharCode.apply(null,
+            new Uint8Array(recordArrayBuffer)));
+      }, () => {
+        record.undecryptable = true;
+        return record;
+      });
+    };
+
+    return {
+      encode,
+      decode
+    };
+  };
+
+We create two Kinto instances, so that we can test syncing from one instance
+to the other, via the public Kinto demo instance on AWS. Creating multiple Kinto
+instances within the same origin is now possible with the recently added
+<code>dbPrefix</code>
+option:
+
+.. code-block:: javascript
+  // Kinto collection:
+  const createCollection = (transformer, testRun, instanceNo) => {
+    const kinto = new Kinto({
+      dbPrefix: `${testRun}-${instanceNo}`,
+      remote: 'https://kinto.dev.mozaws.net/v1/',
+      headers: {
+        Authorization: 'Basic ' + btoa('public-demo:s3cr3t')
+      }
+    });
+
+    return kinto.collection(`kinto-encryption-example-${testRun}`, {
+      remoteTransformers: [ transformer ]
+    });
+  };
+
+  var coll1, coll2;
+  const prepare = () => {
+    return generateAesKey().then(aesKey => {
+      return createTransformer(aesKey);
+    }).then(transformer => {
+      // Create two fresh empty Kinto instances for testing:
+      const testRun = new Date().getTime().toString();
+      coll1 = createCollection(transformer, testRun, 1);
+      coll2 = createCollection(transformer, testRun, 2);
+    });
+  };
+
+Now, let's test if we can create an item in collection 1, sync it up (the data
+should be encrypted on the wire, we can check that on the Network tab of the
+browser console):
```

Comment:
Side note, adding a screenshot of the network tab in action would be super nice.